rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/custom_client_info.hpp

```diff
@@ -100,19 +102,19 @@ class ClientListener : public eprosima::fastrtps::SubscriberListener
     if (conditionMutex_ != nullptr) {
       std::unique_lock<std::mutex> clock(*conditionMutex_);
       if (!list.empty()) {
-        response = list.front();
+        response = std::move(list.front());
```

Comment:
this move is necessary because of the unique_ptr inside the response, right? This is potentially dangerous because the list is afterwards invalid.
If you wouldn't call `pop_front()` immediately afterwards (which you do!), and try to access the first element of the list again, this will segfault.

Reply:
210346b

I don't see an API in the list class that combines getting the first element and removing it. 
210346b  uses a lambda so at least all the risky stuff is in one spot. It should be fine as long as the move constructor doesn't throw an exception.